src/sql/base/browser/ui/modal/modal.ts

```diff
@@ -305,17 +305,21 @@ export abstract class Modal extends Disposable implements IThemable {
 			}
 		});
 		this._resizeListener = DOM.addDisposableListener(window, DOM.EventType.RESIZE, (e: Event) => {
-			this.layout(DOM.getTotalHeight(this._builder.getHTMLElement()));
+			this._layout();
 		});
 
-		this.layout(DOM.getTotalHeight(this._builder.getHTMLElement()));
+		this._layout();
 		TelemetryUtils.addTelemetry(this._telemetryService, TelemetryKeys.ModalDialogOpened, { name: this._name });
 	}
 
 	/**
 	 * Required to be implemented so that scrolling and other functions operate correctly. Should re-layout controls in the modal
 	 */
-	protected abstract layout(height?: number): void;
```

Comment:
>layout [](start = 20, length = 6)

When should be call layout VS _layout

Reply:
_layout is basically the internal method to tell the modal to layout itself. It gets the sizes and then calls layout() with the correct dimension.